cbioportal_etl/scripts/diff_studies.py

```diff
@@ -419,65 +436,68 @@ def run_py(args):
         current_attr_keys -= set(comparisons[comp]["attr_skip"])
         current_attr_keys |= set(comparisons[comp]["attr_implicit"])
 
-        # Pull current data from portal URL; build update data from file
+        # Pull current data from portal URL
         current_data: dict = data_clinical_from_study(
             url=formatted_url,
             auth_headers=auth_headers,
             study_id=args.study,
             data_type=comp,
             aggr_list=aggregate_vals,
         )
-        update_data, update_attr_keys, file_header, file_as_list, big_head = table_to_dict(
-            in_file=comparisons[comp]["data_table"],
-            key=comparisons[comp]["data_table_key"],
-            aggr_list=aggregate_vals,
+
+        # build update data from file
+        update_header, update_columns, update_body = parse_file(
+            file_path=comparisons[comp]["data_table"], header_symbol="#", column_names=True,
         )
 
-        # Get attrs/ids uniq to the current and update datasets
+        # Build update_attr_keys from column names
+        update_attr_keys: set[str] = set(update_columns)
+        update_attr_keys.remove(comparisons[comp]["data_table_key"])
+
+        # Get attrs uniq to the current and update datasets
         current_only_attr: set[str]
         update_only_attr: set[str]
         shared_attr: set[str]
         current_only_attr, update_only_attr, shared_attr = get_set_venn(
             left=current_attr_keys, right=update_attr_keys
         )
-        current_only_ids: set[str]
-        update_only_ids: set[str]
-        shared_ids: set[str]
-        current_only_ids, update_only_ids, shared_ids = get_set_venn(
-            left=set(current_data.keys()), right=set(update_data.keys())
+
+        delta_data, delta_lines, update_only_data, update_only_lines = split_data_file(
+            data_lines=update_body,
+            column_names=update_columns,
+            key=comparisons[comp]["data_table_key"],
+            aggr_list=aggregate_vals,
+            current_data=current_data,
+            shared_attrs=shared_attr,
         )
-        # Get and print ids that are present in both but have had updated attrs
-        with open(f"{comp.lower()}_current_v_update.txt", "w") as diff_out:
-            update_delta_ids, update_delta_attr_cts = catalog_and_print_delta_ids(
+
+        # Get ids uniq to the current and update datasets
+        update_only_ids: set[str] = set(update_only_data.keys())
+        shared_ids: set[str] = set(shared_data.keys())
+        current_only_ids: set[str] = set(current_data.keys()) - shared_ids
+
+        # report before and after for deltas
+        delta_attr_count: dict[str, int] = {}
+        if delta_data:
+            delta_attr_count = print_and_count_delta_attr(
+                clin_type=clin_type,
```

Comment:
IDE reports:
`delta_attr_count` not use, `clin_type` not defined, and `delta_data`
```
Argument of type "dict[str, dict[str, str]]" cannot be assigned to parameter "delta_data" of type "dict[str, str]" in function "print_and_count_delta_attr"
  "dict[str, dict[str, str]]" is not assignable to "dict[str, str]"
    Type parameter "_VT@dict" is invariant, but "dict[str, str]" is not the same as "str"
    Consider switching from "dict" to "Mapping" which is covariant in the value typePylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
```

Reply:
Got it.